src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -99,160 +96,223 @@ def _build_aligned_seq(
+        # a different query.
+        if self.type == ScheduledQueryType.SERIES:
+            updated_metrics_query = self._initialize_series(updated_metrics_query)
+        elif self.type == ScheduledQueryType.TOTALS:
+            updated_metrics_query = self._initialize_totals(updated_metrics_query)
+
+        # We recursively apply the initialization transformations downstream.
+        updated_next = None
+        if self.next is not None:
+            updated_next = self.next.initialize(
+                organization, projects, blocked_metrics_for_projects
+            )
 
-            # In case we have more than one filter, we have to group them into an `AND`.
-            if len(inner_snuba_filters) > 1:
-                groups_filters.append(BooleanCondition(BooleanOp.AND, inner_snuba_filters))
-            else:
-                groups_filters.append(inner_snuba_filters[0])
+        return replace(self, metrics_query=updated_metrics_query, next=updated_next)
 
-        # In case we have more than one filter, we have to group them into an `OR`.
-        if len(groups_filters) > 1:
-            groups_filters = [BooleanCondition(BooleanOp.OR, groups_filters)]
+    def _initialize_series(self, metrics_query: MetricsQuery) -> MetricsQuery:
+        updated_metrics_query = metrics_query
 
-        merged_query = TimeseriesConditionInjectionVisitor(groups_filters).visit(
-            self.metrics_query.query
-        )
-        return replace(
-            self,
-            metrics_query=self.metrics_query.set_query(merged_query),
+        # A series query runs always up to the maximum query limit.
+        updated_metrics_query = updated_metrics_query.set_limit(SNUBA_QUERY_LIMIT)
+
+        return updated_metrics_query
+
+    def _initialize_totals(self, metrics_query: MetricsQuery) -> MetricsQuery:
+        updated_metrics_query = metrics_query
+
+        # A totals query doesn't have an interval.
+        updated_metrics_query = updated_metrics_query.set_rollup(
+            replace(updated_metrics_query.rollup, interval=None, totals=True)
         )
 
-    def filter_blocked_projects(
-        self,
+        if self.order:
+            updated_metrics_query = updated_metrics_query.set_rollup(
+                replace(updated_metrics_query.rollup, orderby=self.order.to_snuba_order())
+            )
+
+        if self.limit:
+            updated_metrics_query = updated_metrics_query.set_limit(self.limit)
+        else:
+            updated_metrics_query = updated_metrics_query.set_limit(SNUBA_QUERY_LIMIT)
+
+        return updated_metrics_query
+
+    def is_empty(self) -> bool:
+        return not self.metrics_query.scope.org_ids or not self.metrics_query.scope.project_ids
+
+    @classmethod
+    def _filter_blocked_projects(
+        cls,
+        metrics_query: MetricsQuery,
         organization: Organization,
-        projects: set[Project],
+        projects: Sequence[Project],
```

Comment:
There was no need to have a `set` since we deduplicated inside the function. Maybe it was a mistake left from before.